### DISCUSSION TITLE
Set Item Duplication Fields automatically

### DISCUSSION THREAD

**Discussion by formfcw** [directus/directus#20024] [Feature Requests]
### Summary

Provide a default behavior for Item Duplication Fields.

### Motivation

Setting Item Duplication Fields for each individual collection is definitely not the most pleasant part of working with Directus in my opinion. It is necessary to have this feature, but most of the time we set the values according to the same principle. I think there is a pattern that we could apply as a default behavior for duplication.

#### What are Item Duplication Fields and what are they for?

Item Duplication Fields can be found at the bottom of your collection's data model settings.

<details><summary>Show example image for Item Duplication Fields</summary>

![Item Duplication Fields](https://github.com/directus/directus/assets/78852214/baf899ef-335a-4cc3-938f-fcd36c166810)

</details>

You can duplicate items of a collection by opening the save options and selecting “Save as Copy”. As soon as you press this button, all the values of the fields that are selected in the “Item Duplication Fields” (see example image above) will be duplicated.

<details><summary>Show screenshot of save options</summary>

![Save Menu](https://github.com/directus/directus/assets/78852214/dc77993d-0173-42dd-acb1-5f0066cf0488)

</details>

In the example above only the `title` and `description` fields are copied to a new item. This seems simple enough, but when you have to delve deep into nested relational fields, it can get really time-consuming.

### Detailed Design

Instead of disabling all the Item Duplication Fields by default, we could apply a pattern that covers most use cases automatically.

#### The pattern

Each collection should duplicate every field of its items by default, except the following:

-   the primary key field
-   system fields (where it makes sense) like `status`, `sort`, `date_created`, `user_created`, `date_updated`, `user_updated`
-   For **relational fields** (M2O, O2M, M2M, M2A, …) there are two cases that should be distinguished and they should be applied recursively:
    -   **Create**: If the field interface only allows the creation of new items, its collection should follow the same duplication rules as the parent collection (no primary key field, standard fields …, two cases for relational fields).
    -   **Select OR Select and Create**: Once the field interface allows the selection of existing items, only the primary key of the related item(s) should be copied. This means that we refer to exactly the same related item(s).

Once we need a different behavior, we would set the item duplication fields manually.

#### Mockup

This is how it could look like:

<img width="724" alt="Item Duplication Fields" src="https://github.com/directus/directus/assets/78852214/d18730b1-6775-4aa5-8170-7c2839753fcc">

… and here is a little legend:

<img width="724" alt="Legend" src="https://github.com/directus/directus/assets/78852214/3e35984f-40d7-45d0-a96e-3db21bcabe2c">

Thus, each checkbox of the item duplication fields could have three states that we can toggle between: `enabled`, `disabled` and `auto`.

#### Must Have:

-   An algorithm that covers most use cases, as described above
-   An option to set the values manually

#### Should Have:

-   An option to combine automatically and manually set fields

#### Could Have:

-   An option to set a default duplication value for an interface (not field) in the interface settings

#### Won't Have:

-   Automatic settings only

### Drawbacks

It’s hard to find one, because it would simplify the process a lot. Maybe it’s not easy to understand and we have to learn it first, but I think the current solution isn’t easy to understand either.

### Alternatives

The impact of not doing this: It will continue to take more time than necessary setting item duplication fields.

### Adoption Strategy

Automatic Item Duplication Fields should apply only to newly created collections (to be on the safe side). But for existing collections we can toggle through the checkboxes states and make them automatic that way.

**Comment by ghost** [comment]
Love this idea! It would help a lot.
